### PR TITLE
feat [breaking changes]: Extensible VirtualMachineError and removed PartialEq trait

### PR DESCRIPTION
**Pull request by hurrikaanig**
# Extensible VirtualMachineError and removed PartialEq trait

## Description

Linked to this issue #751 I added VirtualMachineError::Other(anyhow::Error) to allow to returing custom errors when using cairo-rs

Also I removed PartialEq trait from VirtualMachineError enum as I found it was a bad practice through my research with @tdelabro help

See: https://users.rust-lang.org/t/help-understanding-io-error-and-lack-of-partialeq/13212

std::io::Error does not implement PartialEq (https://doc.rust-lang.org/std/io/struct.Error.html)

I have replaced assert_eq!() in tests with assert!(matches!()) (https://stackoverflow.com/questions/57234140/how-to-assert-io-errors-in-rust)

## Checklist
- [x] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/hint_processor/builtin_hint_processor/blake2s_utils.rs [hunk 1]
```diff
@@ -245,12 +245,12 @@ mod tests {
         //Create hint data
         let ids_data = ids_data!["output"];
         //Execute the hint
-        assert_eq!(
+        assert!(matches!(
```
Comment (tdelabro) [inline src/hint_processor/builtin_hint_processor/blake2s_utils.rs:248]: use assert_matches! instead: https://crates.io/crates/assert_matches

File: src/vm/errors/vm_exception.rs [hunk 2]
```diff
@@ -307,17 +307,17 @@ mod test {
         let program =
             program!(instruction_locations = Some(HashMap::from([(pc, instruction_location)])),);
         let runner = cairo_runner!(program);
-        let vm_excep = VmException {
+        let _vm_excep = VmException {
```
Comment (tdelabro) [inline src/vm/errors/vm_exception.rs:310]: This variable is used (l319). It shouldn't be named with an underscore prefixed name.

File: src/vm/runners/builtin_runner/mod.rs [hunk 3]
```diff
@@ -1058,10 +1058,13 @@ mod tests {
             mayberelocatable!(0, 4).into(),
         ]];
 
-        assert_eq!(
-            builtin.run_security_checks(&vm),
-            Err(MemoryError::MissingMemoryCellsWithOffsets("bitwise", vec![0],).into()),
-        );
+        let _result = builtin.run_security_checks(&vm);
```
Comment (tdelabro) [inline src/vm/runners/builtin_runner/mod.rs:1061]: `_result` is used later, it shouldn't be underscore prefixed

File: src/vm/runners/builtin_runner/mod.rs [hunk 4]
```diff
@@ -1084,10 +1087,11 @@ mod tests {
             mayberelocatable!(0, 5).into(),
         ]];
 
-        assert_eq!(
-            builtin.run_security_checks(&vm),
-            Err(MemoryError::MissingMemoryCells("bitwise").into()),
-        );
+        let _result = builtin.run_security_checks(&vm);
```
Comment (tdelabro) [inline src/vm/runners/builtin_runner/mod.rs:1090]: `_result` is used later on, it shouldn't be underscore prefixed

File: src/vm/runners/builtin_runner/mod.rs [hunk 5]
```diff
@@ -1103,11 +1107,13 @@ mod tests {
             mayberelocatable!(0, 4).into(),
             mayberelocatable!(0, 5).into(),
         ]];
-
-        assert_eq!(
-            builtin.run_security_checks(&vm),
-            Err(MemoryError::MissingMemoryCellsWithOffsets("hash", vec![0],).into()),
-        );
+        let _result = builtin.run_security_checks(&vm);
```
Comment (tdelabro) [inline src/vm/runners/builtin_runner/mod.rs:1110]: `_result` is used later on, it shouldn't be underscore prefixed

File: src/vm/runners/builtin_runner/mod.rs [hunk 6]
```diff
@@ -1120,10 +1126,11 @@ mod tests {
 
         vm.memory.data = vec![vec![mayberelocatable!(0, 0).into()]];
 
-        assert_eq!(
-            builtin.run_security_checks(&vm),
-            Err(MemoryError::MissingMemoryCells("hash").into()),
-        );
+        let _result = builtin.run_security_checks(&vm);
```
Comment (tdelabro) [inline src/vm/runners/builtin_runner/mod.rs:1129]: `_result` is used later on, it shouldn't be underscore prefixed

File: src/vm/runners/builtin_runner/mod.rs [hunk 7]
```diff
@@ -1143,10 +1150,11 @@ mod tests {
             mayberelocatable!(22).into(),
         ]];
 
-        assert_eq!(
-            builtin.run_security_checks(&vm),
-            Err(MemoryError::MissingMemoryCells("range_check").into()),
-        );
+        let _result = builtin.run_security_checks(&vm);
```
Comment (tdelabro) [inline src/vm/runners/builtin_runner/mod.rs:1153]: `_result` is used later on, it shouldn't be underscore prefixed

File: src/vm/runners/builtin_runner/mod.rs [hunk 8]
```diff
@@ -1157,10 +1165,11 @@ mod tests {
 
         vm.memory.data = vec![vec![None, mayberelocatable!(0).into()]];
 
-        assert_eq!(
-            builtin.run_security_checks(&vm),
-            Err(MemoryError::MissingMemoryCells("range_check").into()),
-        );
+        let _result = builtin.run_security_checks(&vm);
```
Comment (tdelabro) [inline src/vm/runners/builtin_runner/mod.rs:1168]: `_result` is used later on, it shouldn't be underscore prefixed

File: src/vm/runners/builtin_runner/mod.rs [hunk 9]
```diff
@@ -1220,10 +1229,11 @@ mod tests {
         // The values stored in memory are not relevant for this test
         vm.memory.data = vec![vec![mayberelocatable!(0).into()]];
 
-        assert_eq!(
-            builtin.run_security_checks(&vm),
-            Err(MemoryError::MissingMemoryCells("ec_op").into()),
-        );
+        let _result = builtin.run_security_checks(&vm);
```
Comment (tdelabro) [inline src/vm/runners/builtin_runner/mod.rs:1232]: Same. And can't you have somethin like this: 
```rust
 assert!(matches!(
            builtin.run_security_checks(&vm),
            Err::<(), VirtualMachineError>(MemoryError::MissingMemoryCells("ec_op").into()),
        ));
    }
```

File: src/vm/security.rs [hunk 10]
```diff
@@ -90,10 +90,9 @@ mod test {
         let runner = cairo_runner!(program);
         let mut vm = vm!();
 
-        assert_eq!(
-            verify_secure_runner(&runner, true, &mut vm),
-            Err(RunnerError::NoProgBase.into()),
-        );
+        let result = verify_secure_runner(&runner, true, &mut vm);
+        let _error: Result<(), VirtualMachineError> = Err(RunnerError::NoProgBase.into());
```
Comment (tdelabro) [inline src/vm/security.rs:94]: `_error` is used

File: src/vm/security.rs [hunk 11]
```diff
@@ -90,10 +90,9 @@ mod test {
...
+        assert!(matches!(result, _error));
```
Comment (tdelabro) [inline src/vm/security.rs:95]: Can't you do all of this in one single line ?

File: src/vm/security.rs (hunk 10 above)
Comment (tdelabro) [reply src/vm/security.rs:94]: Actual name should be `expected_error`

File: src/vm/errors/vm_errors.rs [hunk 12]
```diff
@@ -143,4 +143,6 @@ pub enum VirtualMachineError {
     NegBuiltinBase,
     #[error("Security Error: Invalid Memory Value: temporary address not relocated: {0}")]
     InvalidMemoryValueTemporaryAddress(Relocatable),
+    #[error("{0}")]
```
Comment (tdelabro) [inline src/vm/errors/vm_errors.rs:146]: You mean to use `    #[error(transparent)]`, see: https://docs.rs/thiserror/latest/thiserror/.
Also you can use `#[from]`, also in the doc. To make type conversion easier.

```rust
#[transparent]
Other(#[from] anyhow::Error),
```

**Review by tdelabro — CHANGES_REQUESTED** [review]
- [x] use assert_matches
- [x] fix variable names
- [ ] try to make your assert in one single line when possible
- [x] improve the `Other` error variant by use of `thiserror` features

File: src/vm/errors/vm_exception.rs (hunk 2 above)
Comment (tdelabro) [reply src/vm/errors/vm_exception.rs:310]: get rid of this temporary value

**Review by tdelabro — APPROVED** [review]

**Comment by Oppen** [comment]
Is there any chance to require the new error to be `PartialEq` instead? Or is the `Any` type restrictive about that?

**Comment by tdelabro** [comment]
@Oppen 

> Is there any chance to require the new error to be PartialEq instead?

It is not something you want to do. It would prevent users to use any error types that do not implement `PartialEq`, and a lot of them don't. An example here is std::io::Error: https://doc.rust-lang.org/std/io/struct.Error.html. It does not implement `PartialEq` so users wouldn't be able to pass this type in the `Other` variant.

The reason why you want `PartialEq` is because it makes writing your tests slightly easier than using `assert_matches!`, but nowhere in the actual executed code, do you need to do this type of comparison, you usually handle your errors using `match`, which is the correct way to do. IMO it's not a good enough reason to impair the lib users' ability to pass their custom error types.

> Or is the `Any` type restrictive about that?

Because the `Other` variant does not have any information about the error by itself, it's the responsibility of the type it contains to carry this information. That's why we want it to implement the trait `Error` (`anyhow::Error` being kind of superset for this trait, adding thread safety, and other nice features. More about it here: https://docs.rs/anyhow/latest/anyhow/struct.Error.html). This way, we can use the `#[transparent]` feature of `thiserror` and display the inner error directly.

In conclusion, there is not really a way around it. If we want an escape variant for users' custom errors, it has to contain an `Error` type, and we cannot expect it to implement `PartialEq` without impairing their ability to pass their custom types.

**Comment by tdelabro** [comment]
See also this comment: https://github.com/lambdaclass/cairo-rs/pull/776#issuecomment-1406281788

**Comment by Oppen** [comment]
> It is not something you want to do. It would prevent users to use any error types that do not implement PartialEq, and a lot of them don't. An example here is std::io::Error: https://doc.rust-lang.org/std/io/struct.Error.html. It does not implement PartialEq so users wouldn't be able to pass this type in the Other variant.

Sad, but we'll live.

File: Cargo.toml [hunk 13]
```diff
@@ -38,6 +38,8 @@ sha2 = {version="0.10.2", features=["compress"]}
 thiserror = "1.0.32"
 generic-array = "0.14.6"
 keccak = "0.1.2"
+anyhow = "1.0.68"
+assert_matches = "1.5.0"
```
Comment (Oppen) [inline Cargo.toml:42]: This should go in `[dev-dependencies]` instead.

File: src/lib.rs [hunk 14]
```diff
@@ -6,3 +6,7 @@ pub mod serde;
 pub mod types;
 pub mod utils;
 pub mod vm;
+
+#[cfg(test)]
+#[macro_use]
+extern crate assert_matches;
```
Comment (Oppen) [inline src/lib.rs:12]: I think this is not necessary if we keep it inside the `mod tests` blocks.

File: src/vm/runners/builtin_runner/keccak.rs [hunk 15]
```diff
@@ -679,7 +679,10 @@ mod tests {
             ((0, 35), 0)
         ];
 
-        let keccak_instance = KeccakInstanceDef::new(2048, vec![1; 8]);
+        let keccak_instance = KeccakInstanceDef {
+            _state_rep: vec![1; 8],
+            ..Default::default()
+        };
```
Comment (Oppen) [inline src/vm/runners/builtin_runner/keccak.rs:685]: I'm not too sure why this change is needed.

File: src/vm/runners/builtin_runner/mod.rs [hunk 16]
```diff
@@ -1022,8 +1022,8 @@ mod tests {
             true,
         ));
         let vm = vm!();
-        // Unused builtin shouldn't fail security checks
-        assert_eq!(builtin.run_security_checks(&vm), Ok(()),);
+        // Unsed builtin shouldnt fail security checks
```
Comment (Oppen) [inline src/vm/runners/builtin_runner/mod.rs:1025]:
```suggestion
        // Unused builtin shouldn't fail security checks
```

File: src/vm/vm_core.rs [hunk 17]
```diff
@@ -3300,19 +3396,21 @@ mod tests {
...
+            )) if x == *String::from("ec_op") &&
+                    y == MaybeRelocatable::Int(felt_str!(
+                        "2739017437753868763038285897969098325279422804143820990343394856167768859289"
+                    )) &&
+                    z == Some(MaybeRelocatable::Int(felt_str!(
+                        "2778063437308421278851140253538604815869848682781135193774472480292420096757"
+                    )))
         );
-        assert_eq!(error.unwrap_err().to_string(), "Inconsistent auto-deduction for builtin ec_op, expected 2739017437753868763038285897969098325279422804143820990343394856167768859289, got Some(Int(2778063437308421278851140253538604815869848682781135193774472480292420096757))");
+        //assert_eq!(error.unwrap_err().to_string(), "Inconsistent auto-deduction for builtin ec_op, expected 2739017437753868763038285897969098325279422804143820990343394856167768859289, got Some(Int(2778063437308421278851140253538604815869848682781135193774472480292420096757))");
```
Comment (Oppen) [inline src/vm/vm_core.rs:3413]: Why are we no longer checking this? Also, why leave the commented test?

File: tests/bitwise_test.rs [hunk 18]
```diff
@@ -19,11 +19,14 @@ fn bitwise_integration_test() {
...
-        cairo_runner.run_until_pc(end, &mut vm, &mut hint_processor) == Ok(()),
+        matches!(
+            cairo_runner.run_until_pc(end, &mut vm, &mut hint_processor),
+            Ok(())
+        ),
         "Execution failed"
     );
     assert!(
-        cairo_runner.relocate(&mut vm,) == Ok(()),
+        matches!(cairo_runner.relocate(&mut vm,), Ok(())),
```
Comment (Oppen) [inline tests/bitwise_test.rs:29]: Use `assert_matches!`.

File: tests/bitwise_test.rs [hunk 19]
```diff
@@ -19,11 +19,14 @@ fn bitwise_integration_test() {
     let mut vm = VirtualMachine::new(true);
     let end = cairo_runner.initialize(&mut vm).unwrap();
     assert!(
-        cairo_runner.run_until_pc(end, &mut vm, &mut hint_processor) == Ok(()),
+        matches!(
```
Comment (Oppen) [inline tests/bitwise_test.rs:22]: Use `assert_matches!`.

File: tests/pedersen_test.rs [hunk 20]
```diff
@@ -1,5 +1,7 @@
 use std::path::Path;
-
+#[cfg(test)]
```
Comment (Oppen) [inline tests/pedersen_test.rs:2]: This file is always running as `#[cfg(test)]`.

File: tests/skip_instruction_test.rs [hunk 21]
```diff
@@ -4,6 +4,9 @@ use cairo_vm::{
     types::program::Program,
     vm::{runners::cairo_runner::CairoRunner, vm_core::VirtualMachine},
 };
+#[cfg(test)]
```
Comment (Oppen) [inline tests/skip_instruction_test.rs:7]: This one as well.

File: tests/struct_test.rs [hunk 22]
```diff
@@ -19,7 +19,10 @@ fn struct_integration_test() {
     let end = cairo_runner.initialize(&mut vm).unwrap();
 
     assert!(
-        cairo_runner.run_until_pc(end, &mut vm, &mut hint_processor) == Ok(()),
+        matches!(
```
Comment (Oppen) [inline tests/struct_test.rs:22]: Use `assert_matches!`.

**Review by Oppen — CHANGES_REQUESTED** [review]
Overall this looks good. There are a few minor comments and we need to update the change log before it can be merged, specially with regards to interface changes.

I'm a bit sad about the added verbosity, but I can't think of any way to avoid it so :shrug:

File: src/vm/errors/vm_errors.rs [hunk 23]
```diff
@@ -143,4 +143,6 @@ pub enum VirtualMachineError {
     NegBuiltinBase,
     #[error("Security Error: Invalid Memory Value: temporary address not relocated: {0}")]
     InvalidMemoryValueTemporaryAddress(Relocatable),
+    #[error(transparent)]
+    Other(#[from] anyhow::Error),
```
Comment (fmoletta) [inline src/vm/errors/vm_errors.rs:147]: Can we use `Box<dyn Error>` here?

File: src/vm/runners/builtin_runner/keccak.rs (hunk 15 above)
Comment (tdelabro) [reply src/vm/runners/builtin_runner/keccak.rs:685]: it was a clippy error.  It had been fixed on main, since, he will rebase and it won't appear anymore

File: src/vm/errors/vm_errors.rs (hunk 23 above)
Comment (tdelabro) [reply src/vm/errors/vm_errors.rs:147]: Yes we can. Why do you prefer it?

Comment (tdelabro) [reply src/vm/errors/vm_errors.rs:147]: https://docs.rs/anyhow/latest/anyhow/struct.Error.html
Anyhow provide a few more interesting features in comparison, especially requiring the error type to be Send and Sync.
What is the rationale for not wanting those?

Comment (fmoletta) [reply src/vm/errors/vm_errors.rs:147]: I dont think it would be a good idea to force users to use the `anyhow` crate if they want to use their custom error types. If `Send` and `Sync` are a necessity, we can always use `Box<dyn Error + Send + Sync>`.

Comment (tdelabro) [reply src/vm/errors/vm_errors.rs:147]: Alright. @hurrikaanig is finishing some work for Kakarot and will come back to this one really soon

**Comment by hurrikaanig** [comment]
Review changes applied @Oppen :)

Sorry for the delay

**Comment by tdelabro** [comment]
@fmoletta @Oppen Does it seems mergeable in this current state?

**Review by fmoletta — APPROVED** [review]
Looks Good!

**Comment by tdelabro** [comment]
@Oppen You can resolve the open conversation and merge?

**Comment by Oppen** [comment]
@tdelabro on it, sorry for the delay.

File: src/types/relocatable.rs [hunk 24]
```diff
@@ -374,51 +374,61 @@ pub fn relocate_address(
...
+        //     "Offset 18446744073709551616 exceeds maximum offset value"
+        // );
     }
 
     #[test]
     fn add_usize_to_relocatable() {
         let addr = MaybeRelocatable::RelocatableValue(relocatable!(7, 65));
-        let added_addr = addr.add_int(&Felt::new(2));
-        assert_eq!(Ok(MaybeRelocatable::from((7, 67))), added_addr);
+        let _added_addr = addr.add_int(&Felt::new(2));
```
Comment (Oppen) [inline src/types/relocatable.rs:427]: This is used, shouldn't we remove the underscore?

File: src/types/relocatable.rs [hunk 25]
```diff
@@ -427,19 +437,24 @@ mod tests {
             "800000000000011000000000000000000000000000000000000000000000004",
             16
         ));
-        let added_addr = addr.add_int(&Felt::one());
-        assert_eq!(Ok(MaybeRelocatable::Int(Felt::new(4_i32))), added_addr);
+        let _added_addr = addr.add_int(&Felt::one());
+        assert_matches!(
+            Ok::<MaybeRelocatable, VirtualMachineError>(MaybeRelocatable::Int(Felt::new(4_i32))),
+            _added_addr
```
Comment (Oppen) [inline src/types/relocatable.rs:443]: Here as well.

File: src/types/relocatable.rs [hunk 26]
```diff
@@ -427,19 +437,24 @@ mod tests {
...
+        );
     }
 
     #[test]
     fn add_bigint_to_relocatable_prime() {
         let addr = MaybeRelocatable::RelocatableValue(relocatable!(1, 9));
-        let added_addr = addr.add_int(&felt_str!(
+        let _added_addr = addr.add_int(&felt_str!(
```
Comment (Oppen) [inline src/types/relocatable.rs:450]: Ditto.

File: src/types/relocatable.rs [hunk 27]
```diff
@@ -427,19 +437,24 @@ mod tests {
...
             "3618502788666131213697322783095070105623107215331596699973092056135872020481"
         ));
-        assert_eq!(
-            Ok(MaybeRelocatable::RelocatableValue(relocatable!(1, 9))),
-            added_addr
+        assert_matches!(
+            Ok::<MaybeRelocatable, VirtualMachineError>(MaybeRelocatable::RelocatableValue(
+                relocatable!(1, 9)
+            )),
+            _added_addr
```
Comment (Oppen) [inline src/types/relocatable.rs:457]: Ditto.

File: src/types/relocatable.rs [hunk 28]
```diff
@@ -449,37 +464,44 @@ mod tests {
             "3618502788666131213697322783095070105623107215331596699973092056135872020488"
         ));
         let addr_b = &MaybeRelocatable::from(Felt::new(17_i32));
-        let added_addr = addr_a.add(addr_b);
-        assert_eq!(Ok(MaybeRelocatable::from(Felt::new(24_i32))), added_addr);
+        let _added_addr = addr_a.add(addr_b);
+        assert_matches!(
+            Ok::<MaybeRelocatable, VirtualMachineError>(MaybeRelocatable::from(Felt::new(24_i32))),
+            _added_addr
```
Comment (Oppen) [inline src/types/relocatable.rs:470]: Ditto.

File: src/types/relocatable.rs [hunk 29]
```diff
@@ -449,37 +464,44 @@ mod tests {
...
-        let added_addr = addr_a.add(addr_b);
-        assert_eq!(
-            Ok(MaybeRelocatable::RelocatableValue(relocatable!(7, 17))),
-            added_addr
+        let _added_addr = addr_a.add(addr_b);
+        assert_matches!(
+            Ok::<MaybeRelocatable, VirtualMachineError>(MaybeRelocatable::RelocatableValue(
+                relocatable!(7, 17)
+            )),
+            _added_addr
```
Comment (Oppen) [inline src/types/relocatable.rs:491]: Ditto.

File: src/types/relocatable.rs [hunk 30]
```diff
@@ -449,37 +464,44 @@ mod tests {
...
```
Comment (Oppen) [inline src/types/relocatable.rs:504]: Ditto.

File: src/types/relocatable.rs [hunk 31]
```diff
@@ -490,22 +512,24 @@ mod tests {
...
-        let added_addr = addr_a.add(addr_b);
-        assert_eq!(
-            Ok(MaybeRelocatable::RelocatableValue(relocatable!(7, 14))),
-            added_addr
+        let _added_addr = addr_a.add(addr_b);
+        assert_matches!(
+            Ok::<MaybeRelocatable, VirtualMachineError>(MaybeRelocatable::RelocatableValue(
+                relocatable!(7, 14)
+            )),
+            _added_addr
```
Comment (Oppen) [inline src/types/relocatable.rs:520]: Ditto.

**Review by Oppen — CHANGES_REQUESTED** [review]
Hi @hurrikaanig @tdelabro. There are a few more things that need changing. They are tiny and should be quick and easy to fix. I talked with Timothée and he said it's OK if @Juan-M-V updates the PR with those changes to get them applied sooner. I think we can get this merged today 🎉

File: src/vm/vm_core.rs (hunk 17 above)
Comment (Oppen) [reply src/vm/vm_core.rs]: This part has not been addressed.

File: src/math_utils.rs [hunk 32]
```diff
@@ -530,34 +531,34 @@ mod tests {
...
+        assert_matches!(isqrt(&n.pow(2_u32)), Ok(_n));
     }
 
     #[test]
     fn calculate_isqrt_c() {
         let n = biguint_str!(
             "3618502788666131213697322783095070105623107215331596699973092056135872020481"
         );
-        assert_eq!(isqrt(&n.pow(2_u32)), Ok(n));
+        assert_matches!(isqrt(&n.pow(2_u32)), Ok(_n));
```
Comment (Oppen) [inline src/math_utils.rs:548]: Let's compare the value as well.

File: src/math_utils.rs [hunk 33]
```diff
@@ -530,34 +531,34 @@ mod tests {
...
     }
 
     #[test]
     fn calculate_isqrt_zero() {
         let n = BigUint::zero();
-        assert_eq!(isqrt(&n), Ok(BigUint::zero()));
+        assert_matches!(isqrt(&n), Ok(_n));
```
Comment (Oppen) [inline src/math_utils.rs:554]: Same here.

File: src/types/relocatable.rs [hunk 34]
```diff
@@ -374,51 +374,63 @@ pub fn relocate_address(
...
 
     #[test]
     fn add_bigint_to_int() {
         let addr = MaybeRelocatable::from(Felt::new(7i32));
-        let added_addr = addr.add_int(&Felt::new(2i32));
-        assert_eq!(Ok(MaybeRelocatable::Int(Felt::new(9i32))), added_addr);
+        let _added_addr = addr.add_int(&Felt::new(2i32));
+        assert_matches!(
+            Ok::<MaybeRelocatable, VirtualMachineError>(MaybeRelocatable::Int(Felt::new(9i32))),
+            _added_addr
```
Comment (Oppen) [inline src/types/relocatable.rs:387]: Here.

File: src/types/relocatable.rs [hunk 35]
```diff
@@ -374,51 +374,63 @@ pub fn relocate_address(
...
+        );
     }
 
     #[test]
     fn add_usize_to_int() {
         let addr = MaybeRelocatable::from(Felt::new(7_i32));
-        let added_addr = addr.add_usize(2);
-        assert_eq!(MaybeRelocatable::Int(Felt::new(9_i32)), added_addr);
+        let _added_addr = addr.add_usize(2);
+        assert_eq!(MaybeRelocatable::Int(Felt::new(9_i32)), _added_addr);
```
Comment (Oppen) [inline src/types/relocatable.rs:395]: Here.

File: src/vm/runners/builtin_runner/ec_op.rs [hunk 36]
```diff
@@ -934,7 +934,7 @@ mod tests {
         ];
         let builtin = EcOpBuiltinRunner::new(&EcOpInstanceDef::default(), true);
 
-        let _error = builtin.deduce_memory_cell(&Relocatable::from((3, 6)), &memory);
+        let _expected_error = builtin.deduce_memory_cell(&Relocatable::from((3, 6)), &memory);
         /*assert_eq!(
```
Comment (Oppen) [inline src/vm/runners/builtin_runner/ec_op.rs:938]: This shouldn't be commented out I think. And `_expected_error` should be used.

File: src/math_utils.rs [hunk 37]
```diff
@@ -530,34 +531,34 @@ mod tests {
...
         let n = biguint!(81);
-        assert_eq!(isqrt(&n), Ok(biguint!(9)));
+        assert_matches!(isqrt(&n), Ok(x) if x == biguint!(9));
     }
 
     #[test]
     fn calculate_isqrt_b() {
         let n = biguint_str!("4573659632505831259480");
-        assert_eq!(isqrt(&n.pow(2_u32)), Ok(n));
+        assert_matches!(isqrt(&n.pow(2_u32)), Ok(_n));
```
Comment (Oppen) [inline src/math_utils.rs:540]: Compare here.

**Review by Oppen — CHANGES_REQUESTED** [review]

File: src/math_utils.rs (hunk 37 above)
Comment (Juan-M-V) [reply src/math_utils.rs]: Resolved: https://github.com/lambdaclass/cairo-rs/pull/783/commits/c0d94bae0a150a61d62a4d74032f794901bd60a9

File: src/types/relocatable.rs (hunk 35 above)
Comment (Juan-M-V) [reply src/types/relocatable.rs]: Resolved: https://github.com/lambdaclass/cairo-rs/pull/783/commits/0247165b2291df522a1d7c35bdd0631bbba25ff1

File: src/types/relocatable.rs (hunk 34 above)
Comment (Juan-M-V) [reply src/types/relocatable.rs]: Resolved https://github.com/lambdaclass/cairo-rs/pull/783/commits/0247165b2291df522a1d7c35bdd0631bbba25ff1

File: src/math_utils.rs (hunk 33 above)
Comment (Juan-M-V) [reply src/math_utils.rs]: Resolved: https://github.com/lambdaclass/cairo-rs/pull/783/commits/bce4570218b698c3572735271047e7be9eb7e8d3

File: src/math_utils.rs (hunk 32 above)
Comment (Juan-M-V) [reply src/math_utils.rs]: Resolved: https://github.com/lambdaclass/cairo-rs/pull/783/commits/bce4570218b698c3572735271047e7be9eb7e8d3

**Review by Oppen — APPROVED** [review]
